WeatherDataCalculator.py

```diff
@@ -0,0 +1,82 @@
+
+
+class WeatherDataCalculator:
+    def __init__(self, weather_data_calculations, weather_data):
+        self.weather_data_calculations = weather_data_calculations
+        self.weather_data = weather_data
+
+    def calculate_year_results(self, year):
+        if year in self.weather_data.keys():
+            max_temp, max_month, max_day, min_temp, min_month, min_day, max_humidity, max_humidity_month, max_humidity_day = \
+                self.calculate_temperatures(year)
+            self.weather_data_calculations[year] = {
+                "max_temp": max_temp,
+                "max_month": max_month,
+                "max_day": max_day,
+                "min_temp": min_temp,
+                "min_month": min_month,
+                "min_day": min_day,
+                "max_humidity": max_humidity,
+                "max_humidity_month": max_humidity_month,
+                "max_humidity_day": max_humidity_day
+            }
+
+    def calculate_temperatures(self, year):
+        max_temp = int(list(list(self.weather_data[year].items())[
+                       0][1].items())[0][1]["Max TemperatureC"])
+        max_month = list(self.weather_data[year].keys())[0]
+        max_day = list(list(self.weather_data[year].items())[0][1].keys())[0]
+        min_temp = int(list(list(self.weather_data[year].items())[
+                       0][1].items())[0][1]["Min TemperatureC"])
+        min_month = max_month
+        min_day = max_day
+        max_humidity = int(list(list(self.weather_data[year].items())[
+                           0][1].items())[0][1]["Max Humidity"])
+        max_humidity_month = max_month
+        max_humidity_day = max_day
+
+        for month in self.weather_data[year]:
+            for day in self.weather_data[year][month]:
+                max_temperature = self.weather_data[year][month][day]["Max TemperatureC"]
+                if not max_temperature == "" and int(max_temperature) > max_temp:
```

Comment:
add empty/null check at the time of file reading also make conversion there as well. Ideally, there should be no data type conversion after file reading.